### PR TITLE
instructions for reviewer - Node File IO

### LINKED ISSUE #91: Create "##Instructions for reviewer" for project: Node and File io
Please add a section to the bottom of this project's _index.md file.
The section heading should be EXACTLY "## Instructions for reviewer"

Basically we have to set reviewers up for success by telling them what success looks like and what to pay attention to when reviewing this project submission.

Description:

- [ ] Look at a bunch of old reviews for this project for each cohort

- [ ] Find the common feedback [This should help you get an idea of what instructions to create]

- [ ] Write them out clearly

- [ ] The section heading should be EXACTLY "## Instructions for reviewer"

### PR DESCRIPTION
**Pull request by Sbonelo01**
Related issues: [https://github.com/Umuzi-org/ACN-syllabus/issues/91]

## Description:

What are you up to? Fill us in :)

## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by kaleem99 — CHANGES_REQUESTED** [review]
nice work, but on line 216 pls your grammar is not correct.
"Ensure that the add function the takes a string as an argument."
should be "Ensure that the add function takes a string as an argument."
pls, use Grammarly before pushing.

**Comment by Sbonelo01** [comment]
> nice work, but on line 216 pls your grammar is not correct.
> "Ensure that the add function the takes a string as an argument."
> should be "Ensure that the add function takes a string as an argument."
> pls, use Grammarly before pushing.

Howzit @kaleem99 , There is no Line 216 on this PR. There is also no `add function`

**Review by kaleem99 — CHANGES_REQUESTED** [review]
sorry for the previous review I looked at the wrong file. your instructions looks good but there are a few spelling mistakes like line 91 you incorrectly spelled the word available, pls use Grammarly to check for these small issues. other than that everything else looks good.

**Review by kaleem99 — APPROVED** [review]
instruction looks good.

**Review by AxelBankuna — CHANGES_REQUESTED** [review]
`The fs module is readily available in Node.js, so there wont be any need to have node_modules in the repo.` : Besides the reason stated above, node_modules should not be in repo.

`The load function should print/consol.log the visitor data. ensure that the visitor data is not returned instead.`: typo "consol".

File: content/projects/nodejs/file-io/_index.md [hunk 1]
```diff
@@ -84,4 +84,17 @@ Your load function should also get a bit of an update.
...
-```
\ No newline at end of file
+```
+
+## Instructions for reviewer
+
+
+- Please make sure that the save method saves the data in lower case and that whitespace is replaced by an underscore. This is the only accepted format.
+
+- The use of string literals is highly encouraged for the save method. It's good if a recruit uses them
```
Comment (AxelBankuna) [inline content/projects/nodejs/file-io/_index.md]: This seems more of a tip for the person doing the project than it is for the reviewer.

File: content/projects/nodejs/file-io/_index.md [hunk 2]
```diff
@@ -84,4 +84,17 @@ Your load function should also get a bit of an update.
...
+
+- The load function should print/console.log the visitor data. ensure that the visitor data is not returned instead.
```
Comment (AxelBankuna) [inline content/projects/nodejs/file-io/_index.md]: Start new sentence with uppercase letter.

**Review by AxelBankuna — APPROVED** [review]

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Please use the word "learner" instead of recruit.

This content is used by other code schools, so we ned language that makes sense there. 
And when we work with employed learnerships, such as Telkom, the word "recruit" doesn't apply

**Review by sheenarbw — CHANGES_REQUESTED** [review]
Please dont use the word "recruit". Rather say "learner".

File: content/projects/nodejs/file-io/_index.md [hunk 3]
```diff
@@ -84,4 +84,15 @@ Your load function should also get a bit of an update.
...
 charlie.comments = "Winning!"
 charlie.save() # results in an UPDATE to visitor_3.json
-```
\ No newline at end of file
+```
+
+## Instructions for reviewer
+
+
+- The save method should save the data in lower case and whitespace should be replaced by an underscore. This is the only accepted format.
```
Comment (sheenarbw) [inline content/projects/nodejs/file-io/_index.md]: You mean the file name should be lowercase?

File: content/projects/nodejs/file-io/_index.md [hunk 4]
```diff
@@ -84,4 +84,15 @@ Your load function should also get a bit of an update.
...
+
+- The load function should print/console.log the visitor data. Ensure that the visitor data is not returned instead.
+
+- Edge cases should be catered for. The recruit should make sure that the load function takes in the proper data type.
```
Comment (sheenarbw) [inline content/projects/nodejs/file-io/_index.md]: please say learner not recruit

**Review by sheenarbw — CHANGES_REQUESTED** [review]

File: content/projects/nodejs/file-io/_index.md [hunk 5]
```diff
@@ -1,87 +1,95 @@
...
+
+```
+charlie = load(3)
+charlie.comments = "Winning!"
+charlie.save() # results in an UPDATE to visitor_3.json
+```
+
+## Instructions for reviewer
+
+- The load function should print/console.log the visitor data. Ensure that the visitor data is not returned instead. UNLESS, the learner has taken up the challenge on laid out in the instructions section (instruction 4)
```
Comment (sheenarbw) [inline content/projects/nodejs/file-io/_index.md]: Consider rewording your firsdt point. The english delicious is :P 

Maybe "taken on the challenge laid out..."

File: content/projects/nodejs/file-io/_index.md [hunk 6]
```diff
@@ -1,87 +1,95 @@
...
+
+- Edge cases should be catered for. The learner should make sure that the load function takes in the proper data type.
+
+- The learner should use mocks and spies if/when testing this project - since the project requires interacting with the file system.
```
Comment (sheenarbw) [inline content/projects/nodejs/file-io/_index.md:95]: Please add something about DRY code. People consistently write repetitive code when figuring out the file names. If they have repeated logic then that's bad.

**Review by sheenarbw — CHANGES_REQUESTED** [review]

File: content/projects/nodejs/file-io/_index.md [hunk 7]
```diff
@@ -1,87 +1,99 @@
...
+
+- The load function should print/console.log the visitor data. Ensure that the visitor data is not returned instead. UNLESS, the learner has taken on the challenge laid out in the instructions section (instruction 4)
+
+- Edge cases should be catered for. The learner should make sure that the load function takes in the proper data type.
+
+- The learner should use mocks and spies if/when testing this project - since the project requires interacting with the file system.
+
+- Please pay careful attention to DRY code. How do you do this :
+ 
+ > Imagine that the code produced by the learner needs to be maintained, imagine that some parts of the code will have to change over time. Ideally, changes to the code should be made just once, and that change should affect everything else without going through the file(s) looking to change the same thing.
```
Comment (sheenarbw) [inline content/projects/nodejs/file-io/_index.md:99]: Why different formatting here?

**Review by sheenarbw — APPROVED** [review]
Some changes

**Pull request merged**